lib/knetwork/kdns/dns.go

```diff
@@ -119,6 +120,9 @@ func (s *DnsServer) HandleControllers() {
 	controllerMap := map[string]*RecordController{}
 	defer close(s.readOnlyChan)
 	defer close(s.newOrExistingChan)
+	defer func() {
+		fmt.Println("closing controllers??", s.Domains)
```

Comment:
Maybe log instead of printing directly to stdout? `DnsServer` has a logger

Also, defers run after the function returns. If I'm reading this right - the controllers are closed on line 137, no? Maybe this should log that the controllers have already been closed